picamera2/picamera2.py

```diff
@@ -937,9 +935,6 @@ def configure_(self, camera_config="preview") -> None:
         self.encode_stream_name = camera_config['encode']
         if self.encode_stream_name is not None and self.encode_stream_name not in camera_config:
             raise RuntimeError(f"Encode stream {self.encode_stream_name} was not defined")
-        elif self.encode_stream_name is None:
```

Comment:
Removed this check as encoders don't need to always encode from encode_stream_name